### PR TITLE
fix(idtoken): increase MaxIdleConnsPerHost to 100 in NewClient

### PR DESCRIPTION
**Pull request by quartzmo**
closes: #1744

### CODE REVIEW

File: idtoken/idtoken.go [hunk 1]
```diff
@@ -54,7 +54,9 @@ func NewClient(ctx context.Context, audience string, opts ...ClientOption) (*htt
 	// Skip DialSettings validation so added TokenSource will not conflict with user
 	// provided credentials.
 	opts = append(opts, option.WithTokenSource(ts), internaloption.SkipDialSettingsValidation())
-	t, err := htransport.NewTransport(ctx, http.DefaultTransport, opts...)
+	httpTransport := http.DefaultTransport.(*http.Transport).Clone()
+	httpTransport.MaxIdleConnsPerHost = 100
```
Comment (codyoss) [inline idtoken/idtoken.go:58]: This seems like a large increase. The default when not set is 2 for comparison. I wonder if there is a different way we could enable users to do something like this.

Comment (quartzmo) [reply idtoken/idtoken.go:58]: This is the same default set in [transport/http/dial.go](https://github.com/googleapis/google-api-go-client/blob/main/transport/http/dial.go#L170). See some potentially relevant discussion [here](https://github.com/OJ/gobuster/issues/127).

Comment (quartzmo) [reply idtoken/idtoken.go:58]: @david-gang Do you have any opinion on hardcoding to `100` here, as opposed to flexible configuration or some other value?

Comment (david-gang) [reply idtoken/idtoken.go:58]: @quartzmo 100 is fine for me. In the product i work, we have 100 hardcoded for every http client.

**Review by codyoss — APPROVED** [review]

**Pull request merged**